src/Features/Core/Portable/CodeRefactorings/ExtractMethod/AbstractExtractMethodCodeRefactoringProvider.cs

```diff
@@ -46,36 +50,77 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             CancellationToken cancellationToken)
+        {
+            var actions = ArrayBuilder<CodeAction>.GetInstance();
+            var methodAction = await ExtractMethod(document, textSpan, cancellationToken).ConfigureAwait(false);
+            actions.AddIfNotNull(methodAction);
+
+            var localFunctionAction = await ExtractLocalFunction(document, textSpan, cancellationToken).ConfigureAwait(false);
+            actions.AddIfNotNull(localFunctionAction);
+
+            return actions.ToImmutable();
+        }
+
+        private async Task<CodeAction> ExtractMethod(Document document, TextSpan textSpan, CancellationToken cancellationToken)
         {
             var result = await ExtractMethodService.ExtractMethodAsync(
                 document,
                 textSpan,
+                localFunction: false,
                 cancellationToken: cancellationToken).ConfigureAwait(false);
             Contract.ThrowIfNull(result);
 
             if (result.Succeeded || result.SucceededWithSuggestion)
             {
                 var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
                 var description = documentOptions.GetOption(ExtractMethodOptions.AllowMovingDeclaration) ?
-                                      FeaturesResources.Extract_Method_plus_Local : FeaturesResources.Extract_Method;
+                                      FeaturesResources.Extract_method_plus_local : FeaturesResources.Extract_method;
 
                 var codeAction = new MyCodeAction(description, c => AddRenameAnnotationAsync(result.Document, result.InvocationNameToken, c));
                 var methodBlock = result.MethodDeclarationNode;
 
-                return (codeAction, methodBlock.ToString());
+                return codeAction;
+            }
+
+            return default;
+        }
+
+        private async Task<CodeAction> ExtractLocalFunction(Document document, TextSpan textSpan, CancellationToken cancellationToken)
+        {
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            if (!syntaxFacts.SupportsLocalFunctionDeclaration(syntaxTree.Options))
+            {
+                return default;
+            }
+
+            var localFunctionResult = await ExtractMethodService.ExtractMethodAsync(
+                document,
+                textSpan,
+                localFunction: true,
+                cancellationToken: cancellationToken).ConfigureAwait(false);
+            Contract.ThrowIfNull(localFunctionResult);
+
+            if (localFunctionResult.Succeeded || localFunctionResult.SucceededWithSuggestion)
+            {
+                var codeAction = new MyCodeAction(FeaturesResources.Extract_local_function, c => AddRenameAnnotationAsync(localFunctionResult.Document, localFunctionResult.InvocationNameToken, c));
```

Comment:
do we not need the `ExtractMethodOptions.AllowMovingDeclaration` logic like in the Extract-Method case above?

Reply:
I believe the "AllowMovingDeclaration" option has been hidden for the past couple years. From asking around, I think it might be possible to even remove the option altogether although I feel like it should be done in a separate PR.

Reply:
👍 thanks.